### PR TITLE
fixes #22085 (int type pandas.series.index contains float type key bug)

### LINKED ISSUE #22085: Checking that a pandas.Series.index contains a value
#### Code Sample, a copy-pastable example if possible

```python
# Your code here
import numpy as np
import pandas as pd

some_numbers = np.random.randint(0,4,size=10)
print(some_numbers)
s = pd.Series(some_numbers)
gb = s.groupby(s).size() / len(s)
print(gb)
1.3 in gb
```
#### Problem description
I reported it [here](https://stackoverflow.com/questions/51563433/checking-that-a-pandas-series-index-contains-a-value). The output of the last line is True instead of being False.

#### Expected Output
False

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.2.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.23.3
pytest: 3.2.1
pip: 9.0.1
setuptools: 40.0.0
Cython: 0.26.1
numpy: 1.15.0
scipy: 0.19.1
pyarrow: None
xarray: None
IPython: 6.1.0
sphinx: 1.6.3
patsy: 0.4.1
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.2
feather: None
matplotlib: 2.2.2
openpyxl: 2.4.8
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 0.9.8
lxml: 3.8.0
bs4: 4.6.0
html5lib: 0.999999999
sqlalchemy: 1.1.13
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by a1shadows**
- [x] closes #22085 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry
checks for type using is_float, is_integer_dtype and if key is a float that does not resolve to an equivalent int and Index is int, returns false.

### CODE REVIEW

**Comment by a1shadows** [comment]
tests failed

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -1949,6 +1949,9 @@ def __nonzero__(self):
     def __contains__(self, key):
         hash(key)
         try:
+            if type(key) == float and self.dtype == int:
```
Comment (jreback) [inline pandas/core/indexes/base.py]: use is_float(key) and is_integer_dtype(self.dtype)

**Review by jreback — CHANGES_REQUESTED** [review]
tests and pls add a whatsnew entry for 0.24

**Comment by jreback** [comment]
note that this duplicates #22360

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -1949,6 +1949,9 @@ def __nonzero__(self):
     def __contains__(self, key):
         hash(key)
         try:
+            if is_float(key) and is_integer_dtype(self.dtype):
+                if key != int(key):
+                    return False
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/base.py]: I would put this outside of the try/except block

Comment (a1shadows) [reply pandas/core/indexes/base.py]: done

Comment (yeojin-dev) [reply pandas/core/indexes/base.py]: @jorisvandenbossche I think it can't pass the test!

Comment (jorisvandenbossche) [reply pandas/core/indexes/base.py]: So it fails when key is NaN ?

Comment (yeojin-dev) [reply pandas/core/indexes/base.py]: I think so. The block is needed.

**Comment by yeojin-dev** [comment]
@a1shadows Oh, this is the same as my PR. 😊
I thanks for your help, but I didn't know.

**Comment by a1shadows** [comment]
duplicate PR

**Pull request closed without merging**